### PR TITLE
Initial Identity Transaction Processor.  

### PR DESCRIPTION
**Pull request by agunde406**
This PR includes the first pass a an identity transaction processor that allows you to set roles and policies.  It also includes unit tests and extends sawtooth cli to be able to set roles and policies from the command line. This is the first step needed to be able to implement on-chain permissioning.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,6 +29,7 @@
 /core/sawtooth_core.egg-info/
 /families/settings/sawtooth_settings/protobuf/
 /families/supplychain/python/sawtooth_supplychain/protobuf
+/families/identity/python/sawtooth_identity/protobuf
```
Comment (boydjohnson) [inline .gitignore]: According to how the protogen script has been modified, it appears `/python` shouldn't be there.

File: cli/sawtooth_cli/identity.py [hunk 2]
```diff
@@ -13,14 +13,16 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+from base64 import b64decode
```
Comment (aludvik) [inline cli/sawtooth_cli/identity.py]: These changes should be squashed into the "Add identity to sawtooth cli." commit

**Comment by peterschwarz** [comment]
This should follow a similar model to the settings family where the state values, which are used by the validator, should be in the base `protos` directory.  The transactions which are, in effect, a reference implementation for updating these values can be in the `families/identity/proto`.

**Comment by agunde406** [comment]
protos updated

**Pull request merged**